misc/collect_logs.sh

```diff
@@ -0,0 +1,293 @@
+
+# Custom resources
+$KUBE_CMD get visualinspection --all-namespaces > $DATADIR/get_cr_all_namespaces.txt &
+$KUBE_CMD describe visualinspection --all-namespaces > $DATADIR/describe_cr_all_namespaces.txt &
+$KUBE_CMD describe crd visualinspectionapps.mas.ibm.com > $DATADIR/describe_crd.txt &
+$KUBE_CMD describe visualinspectionapps.mas.ibm.com > $DATADIR/describe_cr.txt &
+$KUBE_CMD describe crd clusterpolicies.nvidia.com > $DATADIR/describe_gpu_crd.txt &
+$KUBE_CMD describe clusterpolicies.nvidia.com > $DATADIR/describe_gpu_cr.txt &
+
+# Get the actual logs from pods
+for pod in $($KUBE_CMD get pods -o name)
+do
+  # OpenShift will append 'pod/' to the beginning of every pod name when we
+  # use the '-o name' option. So, use ${pod:4} when getting the pod name to
+  # strip off the 'pod/' prefix.
+  $KUBE_CMD logs $pod > $DATADIR/logs_${pod:4} &
+done
+
+# Get GPU logs if the GPU namespace exists
+if $KUBE_CMD get project gpu-operator-resources &> /dev/null
+then
+  $KUBE_CMD get deployments --namespace gpu-operator-resources > $DATADIR/nvidia_get_deployments.txt &
+  $KUBE_CMD describe deployments --namespace gpu-operator-resources > $DATADIR/nvidia_describe_deployments.txt &
+  $KUBE_CMD get pods --namespace gpu-operator-resources > $DATADIR/nvidia_get_pods.txt &
+  $KUBE_CMD describe pods --namespace gpu-operator-resources > $DATADIR/nvidia_describe_pods.txt &
+  $KUBE_CMD get daemonsets --namespace gpu-operator-resources > $DATADIR/nvidia_get_daemonsets.txt &
+  $KUBE_CMD describe daemonsets --namespace gpu-operator-resources > $DATADIR/nvidia_describe_daemonsets.txt &
+fi
+
+# If we're running in standalone mode, get system details
+if $STANDALONE
+then
+  info "Standalone mode selected. Collecting system details..."
+
+  # Get distro information
+  if [[ $(cat /etc/os-release 2>/dev/null) =~ Ubuntu ]];          then ubuntu=1; fi
+  if [[ $(cat /etc/os-release 2>/dev/null) =~ Red ]];             then redhat=1; fi
+  if [[ $(cat /etc/os-release 2>/dev/null) =~ Fedora ]];          then redhat=1; fi
+  if [[ $(grep platform /proc/cpuinfo 2>/dev/null) =~ pSeries ]]; then VM=1;     fi
+  ps_fields="user,group,ruser,rgroup,pid,tid,ppid,pgid,uid,tgid,nlwp,sgi_p,nice,pri,etime,pcpu,pmem,lim,rss,s,size,sz,vsz,trs,stat,f,stime,time,tty,comm,args"
+
+  ulimit -a                            >  $DATADIR/ulimits
+  top -n 1 -b                          >  $DATADIR/top
+  ps -Tdaeo $ps_fields                 >  $DATADIR/ps
+  env                                  >  $DATADIR/env
+  uptime                               >  $DATADIR/uptime
+  date                                 >> $DATADIR/uptime
+  lspci -vvv                           >  $DATADIR/lspci_vvv
+  cat /etc/*-release | egrep "PRETTY_NAME|^NAME|VERSION_ID|^VERSION=" >  $DATADIR/codelevels 2>/dev/null
+  uname -a                             >> $DATADIR/codelevels
+  if [[ $ubuntu ]]
+  then
+    dpkg -l|egrep "visual-inspection"   >> $DATADIR/codelevels
```

Comment:
So, this script will not work with any version prior to 1.3.0?

Reply:
Well the rest of the script will work okay, this file will just end up blank.